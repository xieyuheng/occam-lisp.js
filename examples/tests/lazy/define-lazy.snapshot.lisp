(define-lazy (my-if p t f) (if p t f))

(my-if #t 1 2)
(my-if #f 1 2)

(assert-equal (my-if #t 1 2) 1)
(assert-equal (my-if #f 1 2) 2)

(my-if
 #t
 (begin (println "my-if true") 1)
 (begin (println "my-if false") 2))
