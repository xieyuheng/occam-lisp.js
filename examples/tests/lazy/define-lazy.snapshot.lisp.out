1
2
"my-if true"
1
